src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/BinaryOperator.cs

```diff
@@ -10,11 +10,11 @@ namespace Microsoft.Health.Fhir.Core.Features.Search.Expressions
     /// </summary>
     public enum BinaryOperator
     {
-        Equal,
-        GreaterThan,
-        GreaterThanOrEqual,
-        LessThan,
-        LessThanOrEqual,
-        NotEqual,
+        Equal = 0,
+        GreaterThan = 1,
```

Comment:
We may want to add a comment on the dependency between these enum values and DateTimeBoundedRangeRewriter.